aggregator/src/bin/janus_cli.rs

```diff
@@ -497,20 +556,57 @@ mod tests {
             .unwrap_err();
 
         // Run the program logic.
-        super::write_schema(&db_handle.pool()).await.unwrap();
+        super::write_schema(&db_handle.pool(), false).await.unwrap();
 
         // Verify that the schema was written (by running a query that would fail if it weren't).
         ds.run_tx(|tx| Box::pin(async move { tx.get_tasks().await }))
             .await
             .unwrap();
     }
 
+    #[tokio::test]
+    async fn write_schema_dry_run() {
+        let db_handle = ephemeral_db_handle();
+        let ds = db_handle.datastore(RealClock::default());
+
+        ds.run_tx(|tx| Box::pin(async move { tx.get_tasks().await }))
+            .await
+            .unwrap_err();
+
+        super::write_schema(&db_handle.pool(), true).await.unwrap();
+
+        // Verify that no schema was written (by running a query that would fail if it weren't).
+        ds.run_tx(|tx| Box::pin(async move { tx.get_tasks().await }))
+            .await
+            .unwrap_err();
+    }
+
     fn task_hashmap_from_slice(tasks: Vec<Task>) -> HashMap<TaskId, Task> {
         tasks.into_iter().map(|task| (*task.id(), task)).collect()
     }
 
+    async fn setup_provision_tasks_testcase(
```

Comment:
nit: I think `run_provision_tasks_testcase()` might be a better name, since this does run `provision_tasks()` after setting up the temporary file.